### ISSUE TITLE
PropType warnings with React 15.5

### ISSUE THREAD

**Issue by galki** [ReactTraining/react-router#5173]
4.1.1 Router component shows the following warnings with React 15.5.4:

```
Warning: Failed prop type: Calling PropTypes validators directly is not supported by the 'prop-types' package. Use PropTypes.checkPropTypes() to call them.

Warning: Failed context type: Calling PropTypes validators directly is not supported by the 'prop-types' package. Use PropTypes.checkPropTypes() to call them.
```

**Comment by ThomasBrekelmans** [comment]
These warnings do not seem to occur when using react-router in a modern React app.
(Which uses npm/yarn for dependencies and Webpack 2 with Babel 6 for compiling/bundling everything).

The warnings only occur when using the pre-built umd build I got after npm installing react-router 4.1.1. 
(Which I have to do in an older legacy project which uses a custom build system based on RequireJS, that requires me to manually copy dependencies over to a specific folder, instead of using npm.)

After manually cloning react-router and running the build script manually the errors seem to be resolved.

What I did exactly:
`$ git clone git@github.com:ReactTraining/react-router.git`
(`$ cd react-router`)
(`$ npm i -g yarn`)
(`$ cd packages/react-router-dom`)
`npm run build`

For reference, I've included a [zip file](https://github.com/ReactTraining/react-router/files/1041465/react-router-issue-5173-reply2.zip) with the react-router-dom(.min).js files from the umd folder you get after npm installing react-router (4.1.1), and the files I built manually as described above.

After running a diff between the non-minified files in each folder the only things that seem to be changing is something internal for React. So I'm still not sure what actually causes this issue. But atleast it's not present in the newly built files included in the zip file.

The only thing I can imagine is that the version of React that was used to generate the umd files included in the npm install was somehow different than when I built it manually.
I'm currently using the latest version of React (15.5.4 at the time of posting this).

**Comment by timdorr** [comment]
I need to rebuild the UMD release. The version of `prop-types` was broken for UMD builds when I pushed out 4.1.1. It's literally just a rebuild and push, no changes needed.

**Comment by timdorr** [comment]
Whoops, want to leave this open as a reminder to myself :)

**Comment by nathancahill** [comment]
Yay, when this is rebuilt it will clear up the last React 15.5.4 migration warnings for us!

**Comment by jochenberger** [comment]
See also #5045

**Comment by PeteX** [comment]
I followed @ThomasBrekelmans' instructions and published the results at [PeteX/react-router-built](https://github.com/PeteX/react-router-built).  This means you can load the files as:

https://cdn.rawgit.com/PeteX/react-router-built/211d1a40/react-router-dom.js
https://cdn.rawgit.com/PeteX/react-router-built/211d1a40/react-router-dom.min.js

I'll probably delete the repository soon after the next release of react-router, but the files should stick around thanks to rawgit's caching.

My files are not quite the same as @ThomasBrekelmans' and that's probably because I specifically checked out 4.1.1 rather than just building the head.

I hope this helps someone!

**Comment by golddranks** [comment]
@PeteX thanks, it helped me!

**Comment by jochenberger** [comment]
Since there doesn't seem to be a new release on the horizon, could you please publish a rebuilt version as 4.1.2?

**Comment by timdorr** [comment]
I know, I know. I'll try to do it today at some point.

**Comment by oliwen** [comment]
I seem to get a similar issue when using fusebox to build instead of webpack....

![image](https://user-images.githubusercontent.com/5767995/28251586-e1d824ca-6a80-11e7-990f-46456a1219ba.png)

**Comment by nathancahill** [comment]
Please @timdorr 🙏

```
git checkout tags/v4.1.1
npm version 4.1.2
lerna build
npm publish
```

**Comment by timdorr** [comment]
Finally found some time for this. Re-published react-router and react-router-dom as 4.1.2. No changes other than updated bundled deps.

**Comment by PeteX** [comment]
I've just tried the new release, and it works perfectly, thanks.

The build I uploaded to Github is no longer needed so I'll delete that.  As I said, I expect the files to stick around thanks to RawGit's caching, but you should probably move your site to release 4.1.2 to be on the safe side.
